### DISCUSSION TITLE
Different descriptor_string values for two CDC devices.

### DISCUSSION THREAD

**Discussion by TechnikMax** [hathach/tinyusb#2504] [Q&A]
I have a Raspberry Pi Pico project using FreeRTOS.

```//------------- CLASS -------------//
#define CFG_TUD_CDC               2
#define CFG_TUD_MSC               0
#define CFG_TUD_HID               0
#define CFG_TUD_MIDI              0
#define CFG_TUD_VENDOR            0
```

I'm successfully using two CDC Devices, but at the moment they share the same port information, such as name, ID, etc., provided by the `tud_descriptor_string_cb` within my `usb_descriptors.c` file.

How can I provide different values based on the interface ID? The main goal is to differentiate between the two on a PC, without opening a connection to the device, e.g., by prefixing the serial number. One CDC device is for data transfer, the other for debugging.

**Comment by hathach** [comment]
check out the examples/cdc_dual_ports
